source/blender/gpu/intern/gpu_material.c

```diff
@@ -2779,6 +2779,16 @@ float *GPU_lamp_dynpersmat(GPULamp *lamp)
 	return &lamp->dynpersmat[0][0];
 }
 
+float *GPU_lamp_get_viewmat(GPULamp *lamp)
```

Comment:
You should return a `const` pointer to avoid modify `lamp->viewmat`just by getting its pointer.